[grammar index] Fix warnings; prefer listing bold page locators.

godbyk opened the pull request:
**WORK IN PROGRESS &mdash; DO NOT MERGE YET**

Fixes issue #1917.

For this PR I used xindy only for the grammar terms index. But we could use it for the other indices as well.

I think latexmk may still use makeindex for all the indices. I don't know if it can be configured to use xindy for just the grammar index or not. (I don't use latexmk much. I'll have to check the docs.)

godbyk committed:
[grammar index] Fix warnings; prefer listing bold page locators.

jensmaurer commented:
Well, we should definitely get latexmk going. Looks like the "latexmkrc" file needs a setting for "makeindex", pointing to xindy instead. And yes, that will be used for all indexes, not just the grammar index.

godbyk committed:
[index] Use xindy for all indexes.

godbyk committed:
[index] Remove space after 'see'.

godbyk committed:
[index] Switch to xindy-style cross-references.

godbyk committed:
Travis CI: Update xindy commands.

godbyk commented:
I've updated the `latexmkrc` config file so that it runs `xindy`. I also modified it so that it automatically builds with pdfLaTeX and only builds the `std.tex` file. So you can just run `latexmk` with no arguments now and it should do the right thing.

I'm currently fixing up the index cross-references. xindy treats cross-references as first-class objects and will report an error if you refer to a nonexistent index entry. As a consequence of this, the argument to the `see` and `seealso` entries must match the unformatted main index entry to which it refers.  In other words, I'm changing `\index{blah|see{something, else}}` to `\index{blah|see{something!else}}`.  (Alternatively, I can instruct xindy to not check the cross-references. Personally, I think it's probably worth the initial pain as it makes the index syntax more consistent [i.e., always use `entry!subentry`] and helps us find dangling references.)

There are a few other odds and ends I'm going to check up on, too. I'll post progress updates as I go and will tidy up the git commit history before the final merge.

jensmaurer commented:
I've added a "needs rebase" tag, repurposing it as a general "needs more work" thing.

godbyk committed:
[index] Fix more cross-references.

godbyk committed:
[index] More xindy/indexing fixes.
